ir_Lego_PF.cpp

```diff
@@ -44,3 +43,55 @@ void IRsend::sendLegoPowerFunctions(uint16_t data, bool repeat)
 }
 
 #endif // SEND_LEGO_PF
+
+//+=============================================================================
+#if DECODE_LEGO_PF
+bool  IRrecv::decodeLegoPowerFunctions (decode_results *results) {
+	static LegoPfBitStreamEncoder bitStreamEncoder;
+	uint16_t data   = 0;
+	int  offset = 1;
+
+	// Check SIZE
+	if (irparams.rawlen < 2 * (bitStreamEncoder.MESSAGE_BITS) - 1)  return false ;  // exclude STOP pause
+
+	// Check START_BIT Mark/Space
+	if (!MATCH_MARK (results->rawbuf[offset++], bitStreamEncoder.IR_MARK_DURATION - 58)) return false ;

```

Comment:
I startet with your expectation but failed in the first place. It took me some time to figure it out why this isn´t working when using the given functionality.
First: What do we get when we are using the IRrecv functionality (+ = High, - = Low):
```
Encoding  : LEGO_PF
Code      : 8117 (16 bits)
Timing[35]: 
     + 250, - 950     + 250, - 450     + 250, - 250     + 150, - 250
     + 200, - 250     + 200, - 250     + 200, - 250     + 200, - 200
     + 200, - 500     + 250, - 200     + 250, - 250     + 150, - 250
     + 200, - 500     + 200, - 250     + 200, - 500     + 250, - 450
     + 250, - 500     + 200
```
This means: all timing are quantized into a multiplier of 50us.
As of this I thought using your proposed values would be working. But at this point we have to take a look at the MATCH_xxx functions. I expected them to build a threshold on the given value e.g. 315 +- 50 -> a value between 265 and 365 will be accepted. Looking at the code we´ll find this (from MATCH_MARK, extracted from boarddefs.h, IRremoteInt.h, IRremote.cpp):
```
#define USECPERTICK    50

#define MARK_EXCESS    100

#define TOLERANCE       25
#define LTOL            (1.0 - (TOLERANCE/100.))
#define UTOL            (1.0 + (TOLERANCE/100.))

#define TICKS_LOW(us)   ((int)(((us)*LTOL/USECPERTICK)))
#define TICKS_HIGH(us)  ((int)(((us)*UTOL/USECPERTICK + 1)))

int  MATCH_MARK (int measured_ticks,  int desired_us)
{
  bool passed = ((measured_ticks >= TICKS_LOW (desired_us + MARK_EXCESS))
                && (measured_ticks <= TICKS_HIGH(desired_us + MARK_EXCESS)));
  return passed;
}
```
What does this mean? First of all, our given value of 158 - 58 = 100 will be offseted by +100us ending up as 200us. This will then multiplied by the tolerance of 25% ending up with 250us as the upper limit and 150us as the lower limit. After that both values are divided by a USECPERTICK of 50us (upper = 5, lower = 3) which then gets compared to the tics read.
-> A valid mark will be detected for 3 to 5 tics (or 150us to 250us) including the limit values itself.
Looking at the values read by IRrecv the mark timings found are 150, 200 and 250. In this case we got a valid reading. (unfortunately sometime the first mark tends to be as short as 100us which then leading to a failed detection)

After I sent the PR I found an other PR for a PF receiver using this code: [ir_PowerFunctions.cpp](https://github.com/wintersandroid/Arduino-IRremote/blob/master/ir_PowerFunctions.cpp).
This will given a more robust detection using the original PF timings as limits. A disadvantage of this solution is the deeper coupling on the implementation of the time measurement mechanism which normally gets abstracted away by using the MATCH_xxx functions.

Any discussion on this very welcome.